src/overlays/actors/ovl_En_Geg/z_en_geg.c

```diff
@@ -97,87 +146,950 @@ static DamageTable D_80BB3ED4 = {
+}
+
+void EnGeg_Init(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    EnGeg* this = THIS;
+    s32 pad2;
+    s32 sp34[] = {
+        0x0000003E,
+        0x00000F64,
+    };
+
+    if (gSaveContext.weekEventReg[61] & 1) {
+        Actor_MarkForDeath(&this->actor);
+        return;
+    }
+
+    ActorShape_Init(&this->actor.shape, 0.0f, func_800B3FC0, 0.0f);
+    SkelAnime_InitFlex(globalCtx, &this->skelAnime, &D_06011AC8, &D_06012DE0, this->jointTable, this->morphTable, 18);
+
+    Collider_InitCylinder(globalCtx, &this->colliderCylinder);
+    Collider_SetCylinder(globalCtx, &this->colliderCylinder, &this->actor, &sCylinderInit);
+    Collider_InitSphere(globalCtx, &this->colliderSphere);
+    Collider_SetSphere(globalCtx, &this->colliderSphere, &this->actor, &sSphereInit);
+    CollisionCheck_SetInfo2(&this->actor.colChkInfo, &sDamageTable, &sColChkInfoInit);
+
+    if (this->actor.update != NULL) {
+        this->unk_248 = Object_GetIndex(&globalCtx->objectCtx, OBJECT_OF1D_MAP);
+        if (this->unk_248 < 0) {
+            Actor_MarkForDeath(&this->actor);
+        }
+    }
 
-extern ColliderSphereInit D_80BB3E70;
-extern ColliderCylinderInit D_80BB3E9C;
-extern CollisionCheckInfoInit2 D_80BB3EC8;
-extern DamageTable D_80BB3ED4;
+    Effect_Add(globalCtx, &this->unk_4DC, 4, 0, 0, &sp34);
+    thisx->draw = NULL;
+    this->unk_4E0 = 100;
+    this->actor.draw = EnGeg_Draw;
+    this->actionFunc = func_80BB217C;
+}
 
-extern UNK_TYPE D_06004DB0;
-extern UNK_TYPE D_060091A8;
-extern UNK_TYPE D_06012DE0;
+void EnGeg_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnGeg* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Geg/func_80BB16D0.s")
+    Collider_DestroyCylinder(globalCtx, &this->colliderCylinder);
+    Collider_DestroySphere(globalCtx, &this->colliderSphere);
+    Effect_Destroy(globalCtx, this->unk_4DC);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Geg/func_80BB178C.s")
+void EnGeg_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnGeg* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Geg/func_80BB18FC.s")
+    this->actionFunc(this, globalCtx);
+    func_80BB1FCC(this, globalCtx);
+    func_80BB2088(this, globalCtx);
+    func_80BB1C8C(this);
+    func_8013D9C8(globalCtx, &this->unk_238, &this->unk_232, 3);
+    func_80BB1D04(this);
+    func_80BB178C(this, globalCtx);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Geg/func_80BB19C0.s")
+s32 func_80BB3728(s16 arg0, s16 arg1, Vec3f* arg2, Vec3s* arg3, s32 arg4, s32 arg5) {
+    Vec3f sp7C;
+    Vec3f sp70 = D_801D15B0;
+    Vec3s sp68;
+    MtxF sp28;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Geg/func_80BB1B14.s")
+    Matrix_MultiplyVector3fByState(&sp70, &sp7C);
+    Matrix_CopyCurrentState(&sp28);
+    func_8018219C(&sp28, &sp68, 0);
+    *arg2 = sp7C;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Geg/func_80BB1C1C.s")
+    if ((arg4 == 0) && (arg5 == 0)) {
```

Comment:
```suggestion
    if (!arg4 && !arg5) {
```
Both `arg4` and `arg5` look boolean from the code